app/models/game.rb

```diff
@@ -69,21 +69,229 @@ def player_missing?
   end
 
   def determine_check(king)
-    opponents_pieces = pieces.where(color: !king.color)
+    # if king is in check, returns list of pieces that threaten it - will evaluate to true if present
+    # instead of just returning true/false, I have this method return the list of enemy pieces putting the
+    # king in check to try to DRY up the code
+    threatening_pieces = []
+    opponents_pieces = self.pieces.where(color: !king.color, captured: false)
     opponents_pieces.each do |piece|
-      if !piece.valid_move?(king.x_position, king.y_position)
-        self.check_status = 0
-      else
-        self.check_status = 1
-        break
+      if piece.valid_move?(king.x_position, king.y_position)
+        self.update_attributes(check_status: 1)
+        threatening_pieces << piece
       end
     end
+    if threatening_pieces.any?
+      return threatening_pieces
+    end
+    self.update_attributes(check_status: 0)
+    return false
   end
 
   def check?
     check_status == 1 ? true : false
   end
 
+  def puts_king_in_check?(moving_piece, x_dest, y_dest) # move any piece and see if it puts that piece's color king in check
```

Comment:
updated this helper method to see if moving *any* piece will put the king in check, including the king (before it only checked for moving the king)